sycl/include/sycl/types.hpp

```diff
@@ -925,8 +1014,143 @@ template <typename Type, int NumElements> class vec {
       return *ptr;
     }
   }
+#endif // __SYCL_DEVICE_ONLY__
+#else  // __INTEL_PREVIEW_BREAKING_CHANGES
```

Comment:
I'd suggest to change all `ifdef`/`ifndef` to `#if defined()`/`#if !defined()` and update the comments to include `[!]defined` in `#else`/`#endif`.

Reply:
Based on our conversation, I've reorganized the files a bit, hopefully making it a bit easier to read. I've moved the vec implementations out to two files and added an additional injection file ( vec_gen_common.in ) that generates swizzles, binops and rellogops.